clients/web/src/rest.js

```diff
@@ -84,19 +84,25 @@ setStaticRoot(
  * any time the server might throw an exception from validating user input,
  * e.g. logging in, registering, or generally filling out forms.
  *
- * @param path The resource path, e.g. "user/login"
- * @param data The form parameter object.
- * @param [type='GET'] The HTTP method to invoke.
- * @param [girderToken] An alternative auth token to use for this request.
+ * @param {Object} opts Options for the request, most of which will be passed through to `$.ajax`.
+ * @param {string} opts.url The resource path, e.g. "user/login"
+ * @param {string} [opts.method='GET'] The HTTP method to invoke.
+ * @param [opts.data] The query string or form parameter object.
+ * @param {?Function} [opts.error] An error callback, as documented in
+ *        `$.ajax <http://api.jquery.com/jQuery.ajax/>`_, or null. If not provided, this will have a
+ *        default behavior of triggering a `'g:alert'` global event, with details of the error.
+ * @param {string} [opts.girderToken] An alternative auth token to use for this request.
+ * @returns {$.Promise} A jqXHR promise, which resolves and rejects
+ *          `as documented by $.ajax<http://api.jquery.com/jQuery.ajax/#jqXHR>`_.
```

Comment:
I don't know of any way to parse restructured text with esdoc (or jsdoc); however, there are plugins that support markdown.  In this case, you can just use the [`@url`](http://usejsdoc.org/tags-inline-link.html) tag.